decidim-admin/app/commands/decidim/admin/create_participatory_process_admin.rb

```diff
@@ -0,0 +1,51 @@
+# frozen_string_literal: true
+module Decidim
+  module Admin
+    # A command with all the business logic when creating a new participatory
+    # process admin in the system.
+    class CreateParticipatoryProcessAdmin < Rectify::Command
+      # Public: Initializes the command.
+      #
+      # form - A form object with the params.
+      # participatory_process - The ParticipatoryProcess that will hold the
+      #   user role
+      def initialize(form, participatory_process)
+        @form = form
+        @participatory_process = participatory_process
+      end
+
+      # Executes the command. Braodcasts these events:
+      #
+      # - :ok when everything is valid.
+      # - :invalid if the form wasn't valid and we couldn't proceed.
+      #
+      # Returns nothing.
+      def call
+        return broadcast(:invalid) if form.invalid?
+        return broadcast(:invalid) unless user
+
+        create_participatory_process_admin
+        broadcast(:ok)
+      end
+
+      private
+
+      attr_reader :form, :participatory_process
+
+      def create_participatory_process_admin
+        ParticipatoryProcessUserRole.create!(
+          role: :admin,
+          user: user,
+          participatory_process: @participatory_process
+        )
+      end
+
+      def user
+        User.where(
```

Comment:
If we raise then we either rescue and show a nice error, or we let it crash. Instead, I'm checking if the userexists, and if it doesn't i send a `broadcast(:invalid)` so that I can handle the error from the controller and show a nice error message there.